rust/processor/src/processors/account_transactions_processor.rs

```diff
@@ -101,20 +102,24 @@ impl ProcessorTrait for AccountTransactionsProcessor {
         let processing_start = std::time::Instant::now();
         let last_transaction_timestamp = transactions.last().unwrap().timestamp.clone();
 
-        let mut account_transactions = AHashMap::new();
-
-        for txn in &transactions {
-            account_transactions.extend(AccountTransaction::from_transaction(txn));
-        }
-        let mut account_transactions = account_transactions
-            .into_values()
-            .collect::<Vec<AccountTransaction>>();
-
-        // Sort by PK
-        account_transactions.sort_by(|a, b| {
-            (&a.transaction_version, &a.account_address)
-                .cmp(&(&b.transaction_version, &b.account_address))
-        });
```

Comment:
Need to make sure to sort by PK to prevent db insertions from locking the same range of rows.

Reply:
The `account_transaction` is already sorted by version.

and BTW I'm not sure if the sorting actually help,
and BTW before my change we were not sorting by PK, the PK is (account_address, transaction_version), not (transaction_version, account_address)

Reply:
I'm not sure about the frequency of potential deadlocks, but this is a good opportunity to fix it here to sort by actual PK (account_address, transaction_version) :)

Reply:
Why do we think it will cause deadlock? It doesn't make sense.

and specifically for this case, IIUC we do have another index on (transaction_version, account_address), and postgres implementation doesn't really differentiate primary v.s. secondary index (other than enforcing unique and not null)..

Reply:
If there are concurrent tasks writing to the same rows and the writes are unsorted,
1. Task 1 locks row C then tries to lock row A
2. Task 2 locks row A then tries to lock row C

That will result in deadlock

Reply:
they all touch different rows, no?